### PR TITLE
Store function name in URL so we can share the link

### PR DESCRIPTION
**Pull request by kien-ngo**
The main purpose is to help the CS team reference the function in a contract with ease

---

## PR-Codex overview
This PR introduces dynamic loading of contract functions based on URL query parameters in the contract functions panel.

### Detailed summary
- Added `useRouter` from Next.js for URL handling
- Dynamically set selected function based on URL query parameter
- Added logic to set default tab index based on function state mutability
- Implemented navigation to specific function URL query parameter on function selection

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
Having issue with the router.push for the contract intro page
Cannot run that page in local

will have to revisit this PR later

File: src/components/contract-functions/contract-function.tsx [hunk 1]
```diff
@@ -228,9 +229,19 @@ export const ContractFunctionsPanel: React.FC<ContractFunctionsPanelProps> = ({
...
+  const _itemName = router.query.name;
+  const _item = fnsOrEvents.find((o) => o.name === _itemName);
   const [selectedFunction, setSelectedFunction] = useState<
     AbiFunction | AbiEvent
-  >(fnsOrEvents[0]);
+  >(_item ?? fnsOrEvents[0]);
+
+  // Set the active tab to Write or Read depends on the `_item`
+  const _defaultTabIndex =
+    _item && "stateMutability" in _item && _item["stateMutability"] === "view"
```
Comment (jnsdls) [inline src/components/contract-functions/contract-function.tsx:242]: I think this needs to be either `view` or `pure` (both show up in "read")

**Review by jnsdls — CHANGES_REQUESTED** [review]
one thing to double check

File: src/components/contract-functions/contract-function.tsx (hunk 1 above)
Comment (kien-ngo) [reply src/components/contract-functions/contract-function.tsx:242]: thanks! lemme take a look

**Review by jnsdls — APPROVED** [review]

**Pull request merged**